packages/eip-5792-middleware/src/hooks/processSendCalls.ts

```diff
@@ -410,6 +450,81 @@ function validateCapabilities(sendCalls: SendCallsPayload) {
       )}`,
     );
   }
+
+  if (capabilities?.auxiliaryFunds) {
+    validateRequiredAssets(
+      capabilities.auxiliaryFunds,
+      chainId,
+      keyringType,
+      isAuxiliaryFundsSupported,
+    );
+  }
+}
+
+/**
+ * Validates EIP-7682 optional `requiredAssets` parameter.
+ *
+ * docs: {@link https://eips.ethereum.org/EIPS/eip-7682#extended-usage-requiredassets-parameter}
+ *
+ * @param auxiliaryFunds - The auxiliaryFunds param to validate.
+ * @param auxiliaryFunds.optional a
+ * @param auxiliaryFunds.requiredAssets a
+ * @param chainId - The chain ID of the incoming request.
+ * @param keyringType - The type of keyring associated with the account.
+ * @param isAuxiliaryFundsSupported - TODO [ffmcgee]
+ * @throws JsonRpcError if the version is not supported.
+ */
+function validateRequiredAssets(
+  // TODO: [ffmcgee] object param instead of multiple params
+  auxiliaryFunds: {
+    optional?: boolean;
+    requiredAssets?: {
+      address: Hex;
+      amount: Hex; // Amount required, as a hex string representing the integer value in the asset's smallest unit
+      standard: string; // Token standard
+      tokenId?: Hex; // Token ID as a hex string (required for ERC-721 and ERC-1155)
+    }[];
+  },
+  chainId: Hex,
+  keyringType: KeyringTypes,
+  isAuxiliaryFundsSupported: (chainId: Hex) => boolean,
+) {
+  const isSupportedAccount =
+    KEYRING_TYPES_SUPPORTING_7702.includes(keyringType);
+
+  if (!isSupportedAccount) {
+    throw new JsonRpcError(
+      EIP7682ErrorCode.UnsupportedChain, // TODO [ffmcgee] update error code
+      'Unsupported account type',
+    );
```

Comment:
i feel like the error code is actually okay or that it just needs to be updated in spec to say not supported by chain for the account since wallet_getCapabilities returns results per chain against a specific account